base/src/main/java/uk/ac/ebi/atlas/utils/GeneSetEnrichmentClient.java

```diff
@@ -81,16 +80,22 @@ else if(! linesHaveCorrectDimensions(lines)){
     }
 
     private List<String[]> fetchResponse(Species species, Collection<String> bioentityIdentifiers) {
-        Reader responseStringReader = new StringReader(
-                restTemplate.getForObject(
-                        MessageFormat.format(
-                                urlPattern,
-                                species.getEnsemblName().toLowerCase(),
-                                Joiner.on(" ").join(bioentityIdentifiers)),
-                        String.class));
-
-        try (TsvReader tsvReader = new TsvReader(responseStringReader)) {
-            return tsvReader.stream().collect(Collectors.toList());
+        try {
+            String response = restTemplate.getForObject(
+                    MessageFormat.format(
+                            urlPattern,
+                            species.getEnsemblName().toLowerCase(),
+                            Joiner.on(" ").join(bioentityIdentifiers)),
+                    String.class);
+
+            Reader responseStringReader = new StringReader(response);
+
+            try (TsvReader tsvReader = new TsvReader(responseStringReader)) {
+                return tsvReader.stream().collect(Collectors.toList());
+            }
+        } catch (RestClientException e) {
+            LOGGER.error(e.getMessage());
```

Comment:
Very minor point: I’d remove the logging here because the error is also being logged in the `catch` block in `fetchEnrichedGenes`.

Reply:
Good spot. Will remove.